hexl/thread-pool/thread-pool-util.hpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <atomic>
+#include <condition_variable>
+#include <functional>
+#include <mutex>
+#include <thread>
+
+namespace intel {
+namespace hexl {
+
+// Enum for thread states
+enum STATE {
+  NONE,      // Undefined state
+  DONE,      // Task is completed and thread is on spin-up
+  KICK_OFF,  // There is a new task to execute, break spin-up
+  RUNNING,   // Executing task
+  SLEEPING,  // Thread is sleeping, waiting for wakeup
+  KILL       // To join thread
+};
+
+// Control variables per thread
+typedef struct s_thread_info {
+  std::atomic_int state{STATE::NONE};
+  std::condition_variable waker;
+  std::mutex wake_mutex;
+  std::thread thread;
+  std::function<void(size_t id, size_t threads)> task;
+} thread_info_t;
```

Comment:
Why is this declared C old school? Also more in keeping shouldn't thread_info_t -> ThreadInfo
```suggestion
struct thread_info_t {
  std::atomic_int state{STATE::NONE};
  std::condition_variable waker;
  std::mutex wake_mutex;
  std::thread thread;
  std::function<void(size_t id, size_t threads)> task;
};
```